app/main.py

```diff
@@ -1,2 +1,153 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 4
+        self.overflow_limit = self.capacity * 2 // 3
+        self.len = 0
+        self.container = [None] * self.capacity
+
+    def resize(self, size: int) -> None:
+        container = self.container
+        self.capacity = size
+        self.overflow_limit = self.capacity * 2 // 3
+        self.container = [None] * size
+        self.len = 0
+        for item in container:
+            if item is not None:
+                self[item[0]] = item[2]
+
+    def _get_key_position(self, key: Any) -> int:
+        key_hash = hash(key) % self.capacity
+        position = key_hash
+
+        def skip_looped_shift(container_row: tuple[Any, int, Any]) -> bool:
+            return container_row[1] > position
+
+        def skip_shifting_rows(container_row: tuple[Any, int, Any]) -> bool:
+            return container_row[1] < key_hash
+
+        def skip_collisions(container_row: tuple[Any, int, Any]) -> bool:
+            return (container_row[1] == key_hash) and \
+                   (container_row[0] != key)
+
+        for checker in (
+                skip_looped_shift,
+                skip_shifting_rows,
+                skip_collisions
+        ):
+            while \
+                    (position < self.capacity) and \
+                    (self.container[position] is not None) and \
+                    checker(self.container[position]):
+                position += 1
+            if position >= self.capacity:
+                position = 0
+                while \
+                        (self.container[position] is not None) and \
+                        checker(self.container[position]):
+                    position += 1
+        return position
```

Comment:
This code is so hard....
Just use only ONE while loop
In this loop check all conditions and increment current index

Reply:
All others code also the same(move_up and move_down)
And so hard.... Please simplify it and remove duplicated code
Also no need to use `\` instead use brackets

Reply:
Remove this code for move_down and move_up as for me it makes code more harder(Or simplify it)